### DISCUSSION TITLE
Importing and using bcrypt in [...nextauth].ts throws a @mapbox error - app dir

### DISCUSSION THREAD

**Discussion by DomicaCode** [nextauthjs/next-auth#6835] [Help]
### Environment

"bcrypt": "^5.1.0",
"next": "13.2.1",
"next-auth": "4.19.2",
"react": "^18.2.0",
"react-dom": "^18.2.0"

### Reproduction URL

https://github.com/DomicaCode/next-auth-example-app-dir

### Describe the issue

Using bcrypt in [...nextauth].ts creates an error. The error specifically in question: 

```
./node_modules/@mapbox/node-pre-gyp/lib/util/nw-pre-gyp/index.html
Module parse failed: Unexpected token (1:0)
You may need an appropriate loader to handle this file type, currently no loaders are configured to process this file. See https://webpack.js.org/concepts#loaders
> <!doctype html>
| <html>
| <head>
```

I think just importing it causes the issue, I've quickly reproduced it by using bcrypt in the `callbacks`.

### How to reproduce

Import and use `bcrypt` in [...nextauth].ts

### Expected behavior

-

**Comment by therealsuji** [comment]
Also, ran into a compile failure while using mongoose as well. 

`
error - node_modules/mongoose/lib/types/objectid.js (21:31) @ prototype
error - TypeError: Cannot read properties of undefined (reading 'prototype')
`

ran into this while following https://next-auth.js.org/adapters/mongodb with brcypt.

**Comment by ericmil87** [comment]
I am having the same issue:

```
Compiling /signin ...
 ⨯ ./node_modules/.pnpm/@mapbox+node-pre-gyp@1.0.11/node_modules/@mapbox/node-pre-gyp/lib/clean.js:8:0
Module not found: Can't resolve 'fs'

https://nextjs.org/docs/messages/module-not-found

Import trace for requested module:
./node_modules/.pnpm/@mapbox+node-pre-gyp@1.0.11/node_modules/@mapbox/node-pre-gyp/lib/ sync ^\.\/.*$
./node_modules/.pnpm/@mapbox+node-pre-gyp@1.0.11/node_modules/@mapbox/node-pre-gyp/lib/node-pre-gyp.js
./node_modules/.pnpm/bcrypt@5.1.1/node_modules/bcrypt/bcrypt.js
./lib/auth.ts
./components/auth/signin-form.tsx
```

**Comment by tayoolaigbe** [comment]
I ran into same issue but using bcryptjs instead of bcrypt fixed mine. I think you should try that.
Make sure you import as:  import bcrypt from 'bcryptjs';

**Comment by ezeparziale** [comment]
I've switched from bcrypt to bcryptjs, and everything is functioning properly.

**Comment by ezeparziale** [comment]
After trying a few times, I got bcrypt to work with auth.js v5.0.0-beta.17.

The trick is to use `bcrypt.compare` in the `auth.ts` file; it works smoothly there. Putting it in `auth.config.ts` causes it to fail.

Make sure to have the auth.ts file within {...authConfig + providers}, and it'll work perfectly with bcrypt.

Check out an example here:: https://github.com/ezeparziale/quark/blob/main/src/auth.ts
